runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -1121,6 +1121,52 @@ static void printClass(TR_OpaqueClassBlock *clazz)
       }
    }
 
+namespace // file-local
+   {
+   class LexicalOrder
+      {
+      enum Result
+         {
+         LESS,
+         EQUAL,
+         GREATER,
+         };
+
+      public:
+      LexicalOrder() : _result(EQUAL) { }
+      LexicalOrder(const LexicalOrder &other) : _result(other._result) { }
+
+      static LexicalOrder by(void *a, void *b) { return LexicalOrder().thenBy(a, b); }
```

Comment:
Should the return value be `LexicalOrder &`?

Reply:
There's no `LexicalOrder` we could reference here that lives past the return

Reply:
I see; then I guess my question is, why have `thenBy` return a reference when passed in `void *` but return by value when passed in `uintptr_t `?

Reply:
Oh, thanks - I didn't mean to copy in `thenBy(uintptr_t, uintptr_t)`

Reply:
Fixed